### PR TITLE
[#EP-816] Validate gift amount is more than $1, less than $10 million…

### PR DESCRIPTION
**Pull request by Omicron7**
…, is a number and no more then 2 decimal places.

### CODE REVIEW

File: src/app/productConfig/productConfigModal.tpl.html [hunk 1]
```diff
@@ -48,14 +48,20 @@ <h4 class="panel-title border-bottom-small" translate>
...
-                         ng-pattern="/^[1-9]\d*(\.\d+)?$/">
+                         ng-min="1"
+                         ng-max="10000000"
+                         ng-pattern="/^[0-9]+(\.[0-9]{1,2})?$/">
                 </div>
               </label>
               <div role="alert" ng-messages="$ctrl.itemConfigForm.amount.$error" ng-if="($ctrl.itemConfigForm.amount | showErrors)">
-                <div class="help-block" ng-message="pattern" translate>Amount must be a valid amount and at least $1</div>
+                <div class="help-block" ng-message="min" translate>Amount must be at least $1.00</div>
+                <div class="help-block" ng-message="max" translate>Amount must be less than $10,000,000.00</div>
```
Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: `Amount cannot be more than $10,000,000.00`? The wording for less than equal is always wierd... ng-max should allow a million flat though right?

Comment (Omicron7) [inline src/app/productConfig/productConfigModal.tpl.html]: Yes, it'll allow the max, but nothing more.

File: src/app/productConfig/productConfigModal.tpl.html [hunk 2]
```diff
@@ -48,14 +48,20 @@ <h4 class="panel-title border-bottom-small" translate>
               <span ng-repeat-end></span>
               <label class="btn btn-default-form u-textLeft custom-amount">
                 <div class="form-group form-group-default u-inline">
-                  <input class="form-control form-control-subtle number" type="text" name="amount"
+                  <input class="form-control form-control-subtle number"
```
Comment (adammeyer) [inline src/app/productConfig/productConfigModal.tpl.html]: input type number can't be used here.  Jason doesn't want the up/down arrows to show on the input, that's why it was using text.

A validation function may need to be written.

Comment (Omicron7) [inline src/app/productConfig/productConfigModal.tpl.html]: so we throw out all the html browsers validation and roll our own?

Comment (OzzieOrca) [inline src/app/productConfig/productConfigModal.tpl.html]: You could look into hiding the spinner with CSS...

**Pull request merged**